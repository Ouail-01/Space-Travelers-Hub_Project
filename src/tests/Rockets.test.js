import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Rocket from '../components/Rockets';

const mockStore = configureMockStore([]);

describe('Rocket', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      rockets: {
        rockets: [
          {
            id: 'falcon1',
            name: 'Falcon 1',
            description: 'A two-stage rocket designed and manufactured by SpaceX.',
            img: 'https://images.example.com/falcon1.jpg',
            reserved: false,
          },
          {
            id: 'falcon9',
            name: 'Falcon 9',
            description: 'A partially reusable two-stage rocket designed and manufactured by SpaceX.',
            img: 'https://images.example.com/falcon9.jpg',
            reserved: true,
          },
        ],
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <Rocket
          id="falcon1"
          name="Falcon 1"
          description="A two-stage rocket designed and manufactured by SpaceX."
          img="https://images.example.com/falcon1.jpg"
          reserved={false}
        />
      </Provider>,
    );
  });

  it('matches snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
