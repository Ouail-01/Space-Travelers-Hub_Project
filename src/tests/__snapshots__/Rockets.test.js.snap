// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket matches snapshot 1`] = `
<div
  className="rocketContainer"
>
  <img
    alt="Falcon 1"
    src="https://images.example.com/falcon1.jpg"
  />
  <div
    className="rocketInfo"
  >
    <h2
      className="rocketTitle"
    >
      Falcon 1
    </h2>
    <p
      className="rocketDesc"
    >
      A two-stage rocket designed and manufactured by SpaceX.
    </p>
    <button
      className="reserveRocket"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
